### PR TITLE
Add shell completion for zsh #1166

### PR DESCRIPTION
**Pull request by Tony-Sol**
Add `brew bundle` completion for zsh #1166 

brew-bundle
brew-bundle-list

For `bash` / `fish` may be added later.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+.idea
+.vscode
+
```
Comment (MikeMcQuaid) [inline .gitignore:3]:
```suggestion
.vscode
```

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks for the PR! Can you split the completions and dump selection into separate PRs? Thanks!

File: completions/zsh/brew-bundle [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+     ex exec
+     e exec
+   )
+   command="${aliases[$command_or_alias]:-$command_or_alias}"
+   print "${command}"
+ }
+
+_brew_bundle_install() {
+  _arguments \
+    '--file[read the Brewfile from this location. Use --file=- to pipe to stdin/stdout]' \
```
Comment (MikeMcQuaid) [inline completions/zsh/brew-bundle:46]: I wonder if there's any way to rely on any of the Homebrew/brew code to autogenerate some/all of this or at least have CI to ensure it's kept in sync?

**Review by MikeMcQuaid — COMMENTED** [review]
Great work so far!

File: .gitignore (hunk 1 above)
Comment (Tony-Sol) [reply .gitignore:3]: @MikeMcQuaid done:)

This PR contains only zsh-completion script

`dump` command logic updated in https://github.com/Homebrew/homebrew-bundle/pull/1217 and should be reviewed and merged after this PR, thanks:)

File: completions/zsh/brew-bundle [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+     d dump
+     clean cleanup
+     cl cleanup
+     rm cleanup
+     ch check
+     ls list
+     l list
+     ex exec
+     e exec
+   )
```
[the remark this answers is not in the document]
Comment (Tony-Sol) [reply completions/zsh/brew-bundle:39]: Oops, sorry, that's is a failed experiment, which i forgot to remove

File: completions/zsh/brew-bundle (hunk 2 above)
Comment (Tony-Sol) [reply completions/zsh/brew-bundle:46]: I didn't research that :c
I know there are some tools to automate this script creation in golang, but not sure for ruby

Comment (MikeMcQuaid) [reply completions/zsh/brew-bundle:46]: No worries, I'll try to help! Check out:

https://github.com/Homebrew/brew/blob/a56a2707075a10827b4c49b3cc51cae7bb94c2d6/completions/zsh/_brew#L4-L6

and

https://github.com/Homebrew/brew/blob/a56a2707075a10827b4c49b3cc51cae7bb94c2d6/Library/Homebrew/completions.rb

which you could maybe use for auto-generation?

Don't worry if you can't handle the e.g. GitHub Actions side: I can help with all that.

Comment (Tony-Sol) [reply completions/zsh/brew-bundle:21]: Well, i'm not good at ruby (actually, it's my first try), but after some research of given code, i think i can make similar for `brew bundle`.
It requires some refactoring and i think, i would make it, if may

Comment (MikeMcQuaid) [reply completions/zsh/brew-bundle:21]: @Tony-Sol just wanted to check how things are going and what you're thinking! Thanks again for the PR!

Comment (Tony-Sol) [reply completions/zsh/brew-bundle:21]: @MikeMcQuaid sorry, i've been unlucky with free time lately :c

i remember about this PR, but can't say, when could continue work on it :c

Comment (MikeMcQuaid) [reply completions/zsh/brew-bundle:21]: @Tony-Sol No worries! Just saw the rebase and wanted to check!

**Comment by MikeMcQuaid** [comment]
@colindean My fear is that without autogen this is going to be a pain to maintain and avoid bitrotting.

**Pull request closed without merging**